app/src/README-eurotoken-payment-request.md

```diff
@@ -0,0 +1,160 @@
+- When you doing a project that is outsourced, you need coding conventions. With the superapp code there were no coding conventions. We did not know where to put our codes and what are the structure. Coding style were also inconsistent throughout the project.
+
+
+## Suggestions:
+- The link needs to be set up with a website which display payment information and methods so that the payer who does not have the EuroToken app can still make use of the EuroToken payment request system.
+
+- Even though our implementation of link sharing is very straight forward, during this progress there is no guarantee that the person behind the public key is the intended benficiary. The suggestion for the future is that, a validation mechanism should be implemented either in the server (with a trusted authority) or user to make sure the link is shared by a receiver linked to the strong identity (for example passport-enrolled identity in the app)
+
+- Implement a multi-payer system to enable multiple payer pay to same user and overview of payers to the payment system
+
+- Tying the payment request to the identity of the requester
+
+- Advertise and update IP and ports of clients to the exchange server so that they can be found there easily
+
+- Implement signature primitives provided by IPv8
+
+- Implement a multi-payer system to enable multiple payer pay to same user and overview of payers to the payment system
+
+- Tying the payment request to the identity of the requester
+
+- Advertise and update IP and ports of clients to the exchange server so that they can be found there easily
+
+
+## Testings:
+
+Tests were not written for Valuetransfer app previously, however we tested our code. There are two types of test you can find in the repository:
+Unit Tests:
+
+- Handling Incoming links
+- Correct Amount input
+- IBAN validation
+
+Integration Tests:
+- Link Signature validation
+
+
+We had difficulties with setting E2E intergration tests since you will need to setup an instance of IPv8. We have also tried to do mock testing, however we came across a variety of problems.
+
+
+
+## Notes
+
+To make the process of link sharing and receiving the transactions easier, we have made some changes to the exchange, namely, we added a method called ```check_payments_done``` in the ```backend/stablecoin/stablecoin.py```. This method checks every 5 seconds for pending transactions to see if they are paid. If so, it will process the payment to finalize the transactions. This is especially relevant when the user uses the second scenario, namely Euro to EuroToken in which a Tikkie link will be used to pay in Euro.
+
+However, this change is not necessary when the server is deployed. When the server is deployed a webhook should be implemented. The webhook will be called by the bank (e.g., in case of Tikkie link, the bank is ABN AMRO) when the user has finished the payment and paid, and then the exchange server can process the transaction.
+
+
+## Notes
```

Comment:
Not sure what happened here but this seems to be an exact duplicate of the paragraph above.